cdap-data-fabric/src/main/java/io/cdap/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -274,19 +261,10 @@ private void undoPartitionDelete(DropPartitionOperation operation) throws Except
     if (srcLocation.exists()) {
       srcLocation.renameTo(files.getLocation(operation.getRelativePath()));
     }
-    // recreating the partition in Hive only makes sense if the rename succeeds
-    addPartitionToExplore(operation.getPartitionKey(), operation.getRelativePath());
   }
 
   private void undoPartitionCreate(AddPartitionOperation operation) throws Exception {
     Exception caughtExn = null;
-    if (operation.isExplorePartitionCreated()) {
-      try {
```

Comment:
Now you have just 1 `catch`, no need to remember exception and do suppressed. Probably even no need to catch at all

Reply:
Removed the try-catch block